src/wp-includes/block-supports/layout.php

```diff
@@ -630,7 +650,16 @@ function wp_render_layout_support_flag( $block_content, $block ) {
 
 	$class_names        = array();
 	$layout_definitions = wp_get_layout_definitions();
-	$container_class    = wp_unique_id( 'wp-container-' );
+
+	/*
+	* We use an incremental ID that is independent per prefix to make sure that
+	* rendering different numbers of blocks doesn't affect the IDs of other
+	* blocks. We need this to make the CSS class names stable across paginations
+	* for features like the enhanced pagination of the Query block.
+	*/
+	$container_class = wp_incremental_id_per_prefix(
+		'wp-container-' . sanitize_title( $block['blockName'] ) . '-layout-'
+	);
```

Comment:
```suggestion
	/*
	 * We use an incremental ID that is independent per prefix to make sure that
	 * rendering different numbers of blocks doesn't affect the IDs of other
	 * blocks. We need this to make the CSS class names stable across paginations
	 * for features like the enhanced pagination of the Query block.
	 */
	$container_class = wp_incremental_id_per_prefix(
		'wp-container-' . sanitize_title( $block['blockName'] ) . '-layout-'
	);
```
Slight alignment issue.

Reply:
Partially fixed in [2d3db44](https://github.com/WordPress/wordpress-develop/pull/5528/commits/2d3db4418b82b8860fd10109cdb9fefaa4b2be09).

See [this comment](https://github.com/WordPress/wordpress-develop/pull/5528#discussion_r1367524573) for the remaining alignment issue.